theano/gpuarray/tests/test_opt.py

```diff
@@ -23,6 +23,16 @@
 from .config import mode_with_gpu, mode_without_gpu, test_ctx_name, SkipTest
 
 
+def _check_stack_trace(thing):
+    def _ops_to_check(op):
+        if not isinstance(op, theano.gof.Op):
+            op = op.op # assume node
+        return not isinstance(op, (theano.compile.ops.Shape_i,
+                                   theano.ifelse.IfElse,
+                                   GpuFromHost, HostFromGpu,
+                                   GpuElemwise))
```

Comment:
Definitely temporary. Probably the reason was that GpuElemwise creates a bunch of nodes in its make_node that it's not obvious how to copy the stack traces to. Also I believe the original Elemwise that is replaced by the optimization does not have a stack trace to begin with.